server/webapp/WEB-INF/rails.new/spec/controllers/api_v1/admin/environments_controller_spec.rb

```diff
@@ -0,0 +1,245 @@
+##########################################################################
+# Copyright 2016 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+require 'spec_helper'
+
+describe ApiV1::Admin::EnvironmentsController do
+
+  describe :index do
+    describe :for_admins do
+      it 'should render a list of environments, for admins' do
+        login_as_admin
+
+        env = BasicEnvironmentConfig.new(CaseInsensitiveString.new('foo-env'))
+        environments = java.util.HashSet.new
+        environments.add(env)
+
+        @environment_config_service = double('environment-config-service')
+        controller.stub(:environment_config_service).and_return(@environment_config_service)
+        @environment_config_service.should_receive(:getEnvironments).and_return(environments)
+
+        get_with_api_header :index
+        expect(response).to be_ok
+        expect(actual_response).to eq(expected_response([env], ApiV1::Config::EnvironmentsConfigRepresenter))
+      end
+    end
+
+    describe :security do
+      it 'should allow anyone, with security disabled' do
+        disable_security
+        expect(controller).to allow_action(:get, :index)
+      end
+
+      it 'should disallow anonymous users, with security enabled' do
+        enable_security
+        login_as_anonymous
+        expect(controller).to disallow_action(:get, :index).with(401, 'You are not authorized to perform this action.')
+      end
+
+      it 'should disallow normal users, with security enabled' do
+        login_as_user
+        expect(controller).to disallow_action(:get, :index).with(401, 'You are not authorized to perform this action.')
+      end
```

Comment:
@jyotisingh @arvindsv — as of now read-only users can see a list of environments and also view the pages for environment variables. Intentional or not — do we want to replicate the same behaviour for API?